### ISSUE TITLE
"Pre" Check requirements for installer

### ISSUE THREAD

**Issue by rallek** [zikula/core#2235]
see also #2144 
I want to see a "pre" Check requirements to check the permissions of app/cache and app/logs.

There might be technical reasons why this is not checked today and throwing an error message. But these error messages should be managed by the installer as well, similar to the Check requirements comming one step later. For users like me it looks like somthing is forgotten in the installer. That would be a major topic in support later and will other go away imediately when trying zikula first time.

I used the tar.gz from ci.zikula.org (build 420). The permissions has been wrong again for cache and logs (755). Other folders have been shown as well 755 in the Check requirements and must be changed. I thought the permissions in a tar.gz should be fine from the beginning.

**Comment by Guite** [comment]
Related to #2162

**Comment by Guite** [comment]
Both `app/cache` and `app/logs` are set to `0777` as part of the CI build here: https://github.com/zikula/core/blob/1.4/build.xml#L147
But after extracting both `tar.gz` and `zip` archives the folders have `0775`. Thus if the web server's user is not in the same group as the user who extracted the archive the following error message appears:
```
Uncaught RuntimeException: Unable to create the cache directory (/home/axel/projects/zikula/src/app/cache/prod)
 in /home/axel/projects/zikula/src/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:586
Stack trace:
#0 /home/axel/projects/zikula/src/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php(510): Symfony\Component\HttpKernel\Kernel-&gt;buildContainer()
#1 /home/axel/projects/zikula/src/lib/Zikula/Bundle/CoreBundle/HttpKernel/ZikulaKernel.php(325): Symfony\Component\HttpKernel\Kernel-&gt;initializeContainer()
#2 /home/axel/projects/zikula/src/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php(133): Zikula\Bundle\CoreBundle\HttpKernel\ZikulaKernel-&gt;initializeContainer()
#3 /home/axel/projects/zikula/src/lib/Zikula/Bundle/CoreBundle/HttpKernel/ZikulaKernel.php(104): Symfony\Component\HttpKernel\Kernel-&gt;boot()
#4 /home/axel/projects/zikula/src/lib/bootstrap.php(32): Zikula\Bundle\CoreBundle\HttpKernel\ZikulaKernel-&gt;boot()
#5 /home/axel/projects/zikula/src/index.ph in <b>/home/axel/projects/zikula/src/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php</b> on line <b>586</b><br />
```

**Comment by craigh** [comment]
```
[Craigs-iMac2013:htdocs/Zikula_Core-1.4.3/app] craig% ls -la
total 240
drwxr-xr-x@ 11 craig  admin    374 Aug 20 17:25 .
drwxr-xr-x@ 27 craig  admin    918 Aug 20 17:29 ..
-rw-r--r--@  1 craig  admin    673 Aug 20 17:25 .htaccess
drwxr-xr-x@  6 craig  admin    204 Aug 20 17:25 Resources
-rw-r--r--@  1 craig  admin   5154 Aug 20 17:25 ZikulaKernel.php
-rw-r--r--@  1 craig  admin   2939 Aug 20 17:25 autoload.php
-rw-r--r--@  1 craig  admin  99619 Aug 20 17:25 bootstrap.php.cache
drwxr-xr-x@  3 craig  admin    102 Aug 20 17:26 cache
drwxr-xr-x@ 11 craig  admin    374 Aug 20 17:25 config
-rw-r--r--@  1 craig  admin   1106 Aug 20 17:25 console
drwxr-xr-x@  3 craig  admin    102 Aug 20 17:26 logs
```

This is `0755` for the directories in question, right?

**Comment by Guite** [comment]
The problem is that it should bw `drwxrwxrwx` instead of `drwxr-xr-x`. Otherwise only the owner has write access.

**Comment by cmfcmf** [comment]
> But after extracting

I assume you used the `-p` flag as stated here? https://github.com/zikula/core/blob/1.4/INSTALL-1.4.md#upload

**Comment by Guite** [comment]
I just verified that the flags are also missing in the Ci job's workspace already.

**Comment by craigh** [comment]
I had **not**. (I had used the Finder's default Archive Utility - see screenshot)

![screen shot 2016-08-22 at 11 43 28 am](https://cloud.githubusercontent.com/assets/350048/17861091/b8819708-685d-11e6-9c30-5c5f5cfd5f5a.png)

So redone...

```
[Craigs-iMac2013:htdocs/Zikula_Core-1.4.3/app] craig% ls -la
total 240
drwxr-xr-x@ 11 craig  admin    374 Aug 22 09:52 .
drwxr-xr-x@ 26 craig  admin    884 Aug 22 09:53 ..
-rw-r--r--@  1 craig  admin    673 Aug 22 09:51 .htaccess
drwxr-xr-x@  6 craig  admin    204 Aug 22 09:51 Resources
-rw-r--r--@  1 craig  admin   5154 Aug 22 09:51 ZikulaKernel.php
-rw-r--r--@  1 craig  admin   2939 Aug 22 09:51 autoload.php
-rw-r--r--@  1 craig  admin  99619 Aug 22 09:52 bootstrap.php.cache
drwxrwxrwx@  3 craig  admin    102 Aug 22 09:53 cache
drwxrwxrwx@ 11 craig  admin    374 Aug 22 09:51 config
-rw-r--r--@  1 craig  admin   1106 Aug 22 09:51 console
drwxrwxrwx@  3 craig  admin    102 Aug 22 09:53 logs
```

so- this is correct!

**Comment by Guite** [comment]
Confirmed: `file:///home/axel/asas/Zikula_Core-1.4.3.build155.tar.gz` worked. I used my file manager's UI before.

**Comment by craigh** [comment]
The checks are already present in the installer here: https://github.com/zikula/core/blob/ce9260a54d368a13351dc9a915f36313ef2df7c7/src/lib/Zikula/Bundle/CoreInstallerBundle/Util/ControllerUtil.php#L112-L112

but this is run AFTER symfony is trying to boot, so the errors will happen. So these checks must be moved BEFORE symfony is attempting to boot up.

**Comment by Guite** [comment]
Here is the stack trace for the record:

```
Uncaught RuntimeException: Unable to create the cache directory (/home/axel/projects/zikula/app/cache/prod)
 in /home/axel/projects/zikula/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:586
Stack trace:
#0 /home/axel/projects/zikula/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php(510): Symfony\Component\HttpKernel\Kernel-&gt;buildContainer()
#1 /home/axel/projects/zikula/lib/Zikula/Bundle/CoreBundle/HttpKernel/ZikulaKernel.php(325): Symfony\Component\HttpKernel\Kernel-&gt;initializeContainer()
#2 /home/axel/projects/zikula/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php(133): Zikula\Bundle\CoreBundle\HttpKernel\ZikulaKernel-&gt;initializeContainer()
#3 /home/axel/projects/zikula/lib/Zikula/Bundle/CoreBundle/HttpKernel/ZikulaKernel.php(104): Symfony\Component\HttpKernel\Kernel-&gt;boot()
#4 /home/axel/projects/zikula/lib/bootstrap.php(32): Zikula\Bundle\CoreBundle\HttpKernel\ZikulaKernel-&gt;boot()
#5 /home/axel/projects/zikula/index.php(14): require('/home/axel/proj. in <b>/home/axel/projects/zikula/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php</b> on line <b>586</b><br />
```

**Comment by craigh** [comment]
closed by #3023
